File: bundle/BarcodeReaderCLI/examples/brcli_example.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.File;

// javac brcli_example.java           ->  compile example
// java -cp . brcli_example           ->  use BarcodeReaderCLI command line 
// java -cp . brcli_example config    ->  use BarcodeReaderCLI configuration file  

public class brcli_example{
	
	public static String implode(String separator, String[] data) {
		StringBuilder sb = new StringBuilder();
		for (int i = 0; i < data.length - 1; i++) {
		//data.length - 1 => to not add separator at the end
			if (!data[i].matches(" *")) {//empty string are ""; " "; "  "; and so on
				sb.append(data[i]);
				sb.append(separator);
			}
		}
		sb.append(data[data.length - 1].trim());
		return sb.toString();
	}	
	
     public static void main(String []args)  throws IOException {
		boolean isWindows = System.getProperty("os.name").toLowerCase().startsWith("windows");
		File myDir = new File(brcli_example.class.getProtectionDomain().getCodeSource().getLocation().getPath());
		
		String exe;
		if (isWindows) exe  = "..\\bin\\BarcodeReaderCLI";
		else      	   exe = "../bin/BarcodeReaderCLI";
		
		String[] cargs = {
		exe,
		"-type=code128",
		"https://wabr.inliteresearch.com/SampleImages/1d.pdf"};

		String configFile = "./brcli-example.config";
		String[] cargsConfig = {
		exe,
		"@" + configFile};
		
		String cmd =  implode(" ", args.length > 0 ? cargsConfig : cargs);
        ProcessBuilder builder = new ProcessBuilder();

		if (isWindows) {
			builder.command("cmd.exe", "/c", cmd);
		} else {
			builder.command("sh", "-c", cmd);
		}
		builder.directory(myDir);
		builder.redirectErrorStream(true);
		Process process = builder.start();
		try (BufferedReader reader = new BufferedReader(
			new InputStreamReader(process.getInputStream()))) {
			String line;
			while ((line = reader.readLine()) != null) {
				System.out.println(line);
			}
		}
     }
}

File: bundle/BarcodeReaderCLI/examples/brcli-example.go
package main

import (
 "os/exec"
 . "fmt"
 "runtime"
 "bytes"
 "os"
)

// go run brcli-example.go           ->  use BarcodeReaderCLI command line 
// go run brcli-example.go config    ->  use BarcodeReaderCLI configuration file  

func main() {
	isWindows := runtime.GOOS == "windows"
	exe, shell, flag := "../bin/BarcodeReaderCLI", "sh", "-c"
	if isWindows {
		exe, shell, flag = "..\\bin\\BarcodeReaderCLI", "cmd.exe", "/c"
	} 

	args := []string{
		exe,
		"-type=code128",
		"https://wabr.inliteresearch.com/SampleImages/1d.pdf"}

	configFile := "./brcli-example.config"
	argsCongig := []string{
		exe,
		"@" + configFile + ""}
	
	cmd := ""
	if (len(os.Args) > 1){
		for _, s := range argsCongig {cmd = cmd + s + " "}
	} else {
		for _, s := range args {cmd = cmd + s + " "}
	}
	 
    proc := exec.Command(shell, flag, cmd)
	var stdout bytes.Buffer
	var stderr bytes.Buffer
	proc.Stdout = &stdout
	proc.Stderr = &stderr
	proc.Run()
	
	if stdout.Len() > 0 { Print("STDOUT: \n" + stdout.String())}
    if stderr.Len() > 0  { Print("STDERR: \n" + stderr.String())}
}